cache/cache.go

```diff
@@ -107,3 +117,95 @@ func (m *MemcachedCache) Set(k string, v []byte, expire int32) {
 func (m *MemcachedCache) Timeouts() uint64 {
 	return atomic.LoadUint64(&m.timeouts)
 }
+
+type ReplicatedMemcached struct {
+	prefix    string
+	instances []*memcache.Client
+
+	// was 50 ms
+	timeoutMs uint64 // timeout for getting data
+	// TODO (grzkv) Scrape this metric
+	timeouts uint64 // timeouts counter
+}
+
+// NewReplicatedMemcached creates a set of identical memcached instances.
+func NewReplicatedMemcached(prefix string, timeout uint64, servers ...string) BytesCache {
+	m := ReplicatedMemcached{
+		prefix:    prefix,
+		timeoutMs: timeout,
+	}
+
+	for _, s := range servers {
+		m.instances = append(m.instances, memcache.New(s))
+	}
+
+	return &m
+}
+
+// Get gets value for the key from the replicated memcached.
+// It sends the request to all replicas and picks the first valid answer
+// (event if it's a not-found) or times out.
```

Comment:
would it be hard to wait for first 'HIT' instead of any valid reply?
A do understand that this increasing a waiting time, but we could compensate this by stricter timeouts for cache queries.

Reply:
Also, this use case means that we make use of inconsistent data on purpose. I would rather ignore that kind of data if possible.

Reply:
Otherwise, it would be nice to have more input. @avereha @gksinghjsr @grlvrl What do you think?

Reply:
I would keep as simple as possible and test.
I guess if "not found" answers are always faster, then we should see this when we restart random memcache instances.
Worst case I can imagine would be restarting memcache in a loop.

Reply:
@avereha would you jeep the current logic?

Reply:
Yes

Reply:
Let's keep it.